### PR TITLE
Add EnumDesktopWindows && stuff

### PR DESCRIPTION
**Pull request by CristiFati**
**To be merged AFTER #2189** (as this is branched from that one).

*EnumDesktopWindows* is the main thing, the rest is just a quick win with minimal code changes.

### CODE REVIEW

File: win32/test/test_win32gui.py [hunk 1]
```diff
@@ -61,5 +64,123 @@ def test_memory_not_writable(self):
...
+        for data in (None, 1, True):
+            self.assertIsNone(win32gui.EnumWindows(self.enum_callback_sle, data))
+        win32api.SetLastError(0)
+        for data in self.default_data_set:
+            self.assertIsNone(win32gui.EnumWindows(self.enum_callback, data))
+        for data in self.default_data_set:
+            self.assertRaises(
+                ValueError, win32gui.EnumWindows, self.enum_callback_exc, data
+            )
+        if sys.version_info[:2] >= (3, 10):
```
Comment (mhammond) [inline win32/test/test_win32gui.py:100]: I don't understand why this only works in 3.10 an up?

**Review by mhammond — COMMENTED** [review]
This seems great but I don't understand the test. Also, could you please add a changelog entry seeing this is a user-visible feature.

File: win32/test/test_win32gui.py (hunk 1 above)
Comment (Avasam) [inline win32/test/test_win32gui.py:100]: Slicing is redundant here

```suggestion
        if sys.version_info >= (3, 10):
```

File: win32/test/test_win32gui.py [hunk 2]
```diff
@@ -61,5 +64,123 @@ def test_memory_not_writable(self):
...
+        win32api.SetLastError(0)
+        for data in self.default_data_set:
+            self.assertRaises(
+                ValueError,
+                win32gui.EnumChildWindows,
+                None,
+                self.enum_callback_exc,
+                data,
+            )
+        if sys.version_info[:2] >= (3, 10):
```
Comment (Avasam) [inline win32/test/test_win32gui.py:125]: Slicing is redundant here
```suggestion
        if sys.version_info >= (3, 10):
```

File: win32/test/test_win32gui.py [hunk 3]
```diff
@@ -61,5 +64,123 @@ def test_memory_not_writable(self):
...
+        for desktop in desktops:
+            for data in self.default_data_set:
+                self.assertRaises(
+                    ValueError,
+                    win32gui.EnumDesktopWindows,
+                    desktop,
+                    self.enum_callback_exc,
+                    data,
+                )
+        if sys.version_info[:2] >= (3, 10):
```
Comment (Avasam) [inline win32/test/test_win32gui.py:173]: Slicing is redundant here
```suggestion
        if sys.version_info[:2] >= (3, 10):
```

File: win32/test/test_win32gui.py (hunk 1 above)
Comment (CristiFati) [reply win32/test/test_win32gui.py]: done

File: win32/test/test_win32gui.py (hunk 2 above)
Comment (CristiFati) [reply win32/test/test_win32gui.py]: done

File: win32/test/test_win32gui.py (hunk 3 above)
Comment (CristiFati) [reply win32/test/test_win32gui.py]: done

File: win32/test/test_win32gui.py (hunk 1 above)
Comment (CristiFati) [reply win32/test/test_win32gui.py]: Good thing you asked, it is just the float value. I return it from the callback and the problem arises when it's converted to long. https://docs.python.org/3/c-api/long.html#c.PyLong_AsLong.

File: CHANGES.txt [hunk 4]
```diff
@@ -160,6 +160,8 @@ Coming in build 307, as yet unreleased
 * Remove outdated and unused remote feature (#2098, @Avasam)
 * Migrated from `distutils` to `setuptools` (#2133, @Avasam)
 
+* Add EnumDesktopWindows (#2219, @CristiFati)
```
Comment (Avasam) [inline CHANGES.txt:163]: I don't think you meant to add this under adodbapi

Comment (CristiFati) [reply CHANGES.txt]: Thanks, it was automatically done by rebase.

**Comment by mhammond** [comment]
> **To be merged AFTER #2189** (as this is branched from that one).

What's the status of that - should it be updated or closed?

**Review by mhammond — APPROVED** [review]
Thanks!

**Comment by CristiFati** [comment]
> > **To be merged AFTER #2189** (as this is branched from that one).
> 
> What's the status of that - should it be updated or closed?

After merging this, I will close it, as this one also contains it.

**Pull request merged**